### PR TITLE
Revise the paragraph style API to include custom ellipsis strings

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: lib/ui/text.dart [hunk 1]
```diff
@@ -374,25 +365,24 @@ class TextStyle {
 // The encoded array buffer has 4 elements.
 //
 //  - Element 0: A bit field where the ith bit indicates wheter the ith element
-//    has a non-null value. Bit 6 indicates whether |fontFamily| is non-null.
-//    Bit 7 indicates whether |fontSize| is non-null. Bit 8 indicates whether
-//    |lineHeight| is non-null. Bit 0 is unused.
```
Comment (Hixie) [inline lib/ui/text.dart]: Shouldn't we keep the documentation about the bits beyond the last element of the list?

File: lib/ui/text.dart [hunk 2]
```diff
@@ -374,25 +365,24 @@ class TextStyle {
 // The encoded array buffer has 4 elements.
 //
 //  - Element 0: A bit field where the ith bit indicates wheter the ith element
```
Comment (Hixie) [inline lib/ui/text.dart]: wheter -> whether

File: lib/ui/text.dart [hunk 3]
```diff
@@ -483,10 +486,11 @@ class ParagraphStyle {
              'textAlign: ${   _encoded[0] & 0x02 == 0x02 ? TextAlign.values[_encoded[1]]    : "unspecified"}, '
              'fontWeight: ${  _encoded[0] & 0x04 == 0x04 ? FontWeight.values[_encoded[2]]   : "unspecified"}, '
              'fontStyle: ${   _encoded[0] & 0x08 == 0x08 ? FontStyle.values[_encoded[3]]    : "unspecified"}, '
-             'textOverflow: ${ _encoded[0] & 0x10 == 0x10 ? TextOverflow.values[_encoded[4]] : "unspecified"}, '
+             'lineCount: ${   _encoded[0] & 0x10 == 0x10 ? _encoded[4]                      : "unspecified"}, '
              'fontFamily: ${  _encoded[0] & 0x20 == 0x20 ? _fontFamily                      : "unspecified"}, '
              'fontSize: ${    _encoded[0] & 0x40 == 0x40 ? _fontSize                        : "unspecified"}, '
-             'lineHeight: ${  _encoded[0] & 0x80 == 0x80 ? "${_lineHeight}x"                : "unspecified"}'
+             'lineHeight: ${  _encoded[0] & 0x80 == 0x80 ? "${_lineHeight}x"                : "unspecified"},'
+             'ellipsis: ${    _encoded[0] & 0x100 == 0x100 ? _ellipsis                      : "unspecified"},'
```
Comment (Hixie) [inline lib/ui/text.dart]: should probably make that `"\"$ellipsis\""`

**Comment by Hixie** [comment]
LGTM, but travis is angry.

@abarth may have opinions too

File: lib/ui/text.dart (hunk 1 above)
Comment (jason-simmons) [inline lib/ui/text.dart]: The documentation was out of date, and it seemed redundant with the code.

File: lib/ui/text.dart (hunk 2 above)
Comment (jason-simmons) [inline lib/ui/text.dart]: done

File: lib/ui/text.dart (hunk 3 above)
Comment (jason-simmons) [inline lib/ui/text.dart]: done

File: lib/ui/text.dart (hunk 1 above)
Comment (Hixie) [inline lib/ui/text.dart]: I'm ok with not calling out the specific bits, but we should at least explain that the zeroth bit is nothing, the 1-Nth bit are about elements 1 through N, and the remaining bits are about the various fields, in some given order.

Comment (jason-simmons) [inline lib/ui/text.dart]: done

**Pull request merged**